File: SimpleSchedulerModels/Schedule.cs
namespace SimpleSchedulerModels;

public record class Schedule(
    long ID, 
    bool IsActive, 
    long WorkerID,
    bool Sunday, 
    bool Monday, 
    bool Tuesday, 
    bool Wednesday, 
    bool Thursday, 
    bool Friday, 
    bool Saturday,
    TimeSpan? TimeOfDayUTC, 
    TimeSpan? RecurTime, 
    TimeSpan? RecurBetweenStartUTC, 
    TimeSpan? RecurBetweenEndUTC,
    bool OneTime
);

File: SimpleSchedulerAngular/src/app/models/schedule.ts
import TimeSpan from "./timespan";

export type Schedule = {
    scheduleID: number,
    isActive: boolean,
    workerID: number,
    sunday: boolean,
    monday: boolean,
    tuesday: boolean,
    wednesday: boolean,
    thursday: boolean,
    friday: boolean,
    saturday: boolean,
    timeOfDayUTC: TimeSpan | null,
    recurTime: TimeSpan | null,
    recurBetweenStartUTC: TimeSpan | null,
    recurBetweenEndUTC: TimeSpan | null,
    oneTime: boolean
};
